### ISSUE TITLE
Unable to install cekit on Fedora 29 as documented

### ISSUE THREAD

**Issue by vorburger** [cekit/cekit#380]
https://cekit.readthedocs.io/en/develop/installation.html#fedora

```
$ sudo dnf install dnf-plugin-copr
Last metadata expiration check: 1:45:10 ago on Tuesday 12 February 2019 09:14:44 PM CET.
No match for argument: dnf-plugin-copr
Error: Unable to find a match
```

**Comment by vorburger** [comment]
```
sudo dnf install python3-cekit
Last metadata expiration check: 1:40:21 ago on Tuesday 12 February 2019 09:14:44 PM CET.
Error: 
 Problem: package python3-cekit-2.2.4-1.noarch requires docker, but none of the providers can be installed
  - package docker-ce-3:18.09.1-3.fc29.x86_64 conflicts with docker provided by docker-2:1.13.1-63.git1185cfd.fc29.x86_64
  - package docker-ce-3:18.09.1-3.fc29.x86_64 conflicts with docker-io provided by docker-2:1.13.1-63.git1185cfd.fc29.x86_64
  - package docker-ce-3:18.09.1-3.fc29.x86_64 conflicts with docker provided by docker-2:1.13.1-62.git9cb56fd.fc29.x86_64
  - package docker-ce-3:18.09.1-3.fc29.x86_64 conflicts with docker-io provided by docker-2:1.13.1-62.git9cb56fd.fc29.x86_64
  - problem with installed package docker-ce-3:18.09.1-3.fc29.x86_64
  - package docker-ce-3:18.09.2-3.fc29.x86_64 conflicts with docker provided by docker-2:1.13.1-63.git1185cfd.fc29.x86_64
  - package docker-ce-3:18.09.2-3.fc29.x86_64 conflicts with docker-io provided by docker-2:1.13.1-63.git1185cfd.fc29.x86_64
  - package docker-ce-3:18.09.2-3.fc29.x86_64 conflicts with docker provided by docker-2:1.13.1-62.git9cb56fd.fc29.x86_64
  - package docker-ce-3:18.09.2-3.fc29.x86_64 conflicts with docker-io provided by docker-2:1.13.1-62.git9cb56fd.fc29.x86_64
  - conflicting requests
  - package python3-cekit-2.2.5-1.noarch requires docker, but none of the providers can be installed
  - package docker-2:1.13.1-61.git9cb56fd.module_2109+7c83ead1.x86_64 is excluded
(try to add '--allowerasing' to command line to replace conflicting packages or '--skip-broken' to skip uninstallable packages)
```

**Comment by vorburger** [comment]
```
$ sudo dnf copr enable @cekit/cekit

You are about to enable a Copr repository. Please note that this
repository is not part of the main distribution, and quality may vary.

The Fedora Project does not exercise any power over the contents of
this repository beyond the rules outlined in the Copr FAQ at
<https://docs.pagure.org/copr.copr/user_documentation.html#what-i-can-build-in-copr>,
and packages are not held to any quality or security level.

Please do not file bug reports about these packages in Fedora
Bugzilla. In case of problems, contact the owner of this repository.

Do you really want to enable copr.fedorainfracloud.org/@cekit/cekit? [y/N]: y
Repository successfully enabled.
```

but on `sudo dnf install python3-cekit` still the same problem as above...

**Comment by vorburger** [comment]
> `$ sudo dnf install dnf-plugin-copr`

it's `dnf-plugins-core` instead of `dnf-plugin-copr` [according to this copr doc](https://docs.pagure.org/copr.copr/how_to_enable_repo.html), so how about #382 ?

**Comment by vorburger** [comment]
>  Problem: package python3-cekit-2.2.4-1.noarch requires docker, but none of the providers can be installed

This is because on the host where I tried this I had [docker's docker-ce](https://docs.docker.com/install/linux/docker-ce/fedora/) installed. I tried to remove that and use Podman's docker CLI emulator (NB #363), but that also didn't work:

```
sudo dnf install podman-docker
(...)
sudo dnf install python3-cekit
Last metadata expiration check: 0:01:35 ago on Wednesday 13 February 2019 02:51:31 AM CET.
Error: 
 Problem: package python3-cekit-2.2.4-1.noarch requires docker, but none of the providers can be installed
  - package podman-docker-1:0.10.1-1.gite4a1553.fc29.noarch conflicts with docker provided by docker-2:1.13.1-63.git1185cfd.fc29.x86_64
  - package podman-docker-1:0.10.1-1.gite4a1553.fc29.noarch conflicts with docker provided by docker-2:1.13.1-62.git9cb56fd.fc29.x86_64
  - problem with installed package podman-docker-1:1.0.0-1.git82e8011.fc29.noarch
  - package podman-docker-1:1.0.0-1.git82e8011.fc29.noarch conflicts with docker provided by docker-2:1.13.1-63.git1185cfd.fc29.x86_64
  - package podman-docker-1:1.0.0-1.git82e8011.fc29.noarch conflicts with docker provided by docker-2:1.13.1-62.git9cb56fd.fc29.x86_64
  - conflicting requests
  - package python3-cekit-2.2.5-1.noarch requires docker, but none of the providers can be installed
  - package docker-2:1.13.1-61.git9cb56fd.module_2109+7c83ead1.x86_64 is excluded
  - package podman-docker-0.8.2.1-1.gitf38eb4f.module_2106+3b446c4f.noarch is excluded
(try to add '--allowerasing' to command line to replace conflicting packages or '--skip-broken' to skip uninstallable packages)
```

**Comment by vorburger** [comment]
So one **_has_** to use `docker-common` to use cekit; system wide, no way to also use Podman...

```
sudo dnf remove docker-ce
sudo dnf remove podman-docker
sudo dnf install python3-cekit
sudo systemctl enable --now docker
```

docker-common is old 1.13.1 vs docker-ce 18.09.1, and (more importantly) shame re. podman-docker.

I don't know much about RPM, but wouldn't be possible to specify only that you just need _a_ `docker` CLI (from whatever package provides it), instead of being bound to and require a specific package?

**Comment by goldmann** [comment]
You mix many things here together.

You refer to documentation (develop branch) that is suited for the upcoming version of Cekit (3.0) but you expect it to be working with Cekit 2.2. Cekit 3.0 has it's dependency mechanism rewritten as part of https://github.com/cekit/cekit/pull/357. See the new documentation: http://docs.cekit.io/en/develop/installation.html#dependencies. If you want to test this, you should use the `python3-cekit-next` package from our Copr repository. You may be interested in https://github.com/cekit/cekit/issues/355 too.

I won't comment on the old vs. new version of Docker.

Regarding Podman, you already found a ticket to add support for it: https://github.com/cekit/cekit/issues/363. This is planned to be implemented for 3.0.

Since your PR #382 was merged, I'm closing this ticket.
